libnetwork/iptables/firewalld.go

```diff
@@ -52,11 +60,46 @@ func firewalldInit() error {
 	if connection, err = newConnection(); err != nil {
 		return fmt.Errorf("Failed to connect to D-Bus system bus: %v", err)
 	}
-	firewalldRunning = checkRunning()
+
+	// If firewalld is not running on the first check poll
 	if !firewalldRunning {
+		ctx, cancel := context.WithTimeout(context.Background(), fireWalldInitializationTimeout)
+		defer cancel()
+
+		for {
+
+			cctx, ccancel := context.WithTimeout(ctx, dbusConnectionTimeout)
+			firewalldRunning, err = func(ctx context.Context) (bool, error) {
+
+				var running bool
+				log.G(ctx).Debugf("Checking status of Firewalld")
+				running, err = checkRunning(ctx)
+
+				if err != nil {
+					return running, err
+				}
+				return running, nil
+			}(cctx)
+			ccancel()
+			if firewalldRunning || err != nil {
+				break
+			}
+			time.Sleep(dbusConnectionTimeout)
+		}
+
+		if err != nil {
+			log.G(context.TODO()).Debugf("Check running failed: %v", err)
+		}
+	}
+
+	if !firewalldRunning {
+		log.G(context.TODO()).Debugf("The dbus API is accessible but firewalld is currently unavailable continuing.")
```

Comment:
I don't think there's a way to get here without `err` being set - and it was logged on line 91. Maybe this log line can be removed? Otherwise, maybe ...
```suggestion
		if err == nil {
			log.G(context.TODO()).Debugf("firewalld is unavailable.")
		}
```